corehq/apps/translations/app_translations/upload_app.py

```diff
@@ -96,6 +109,38 @@ def process_bulk_app_translation_upload(app, workbook, expected_headers, lang=No
     return msgs
 
 
+def get_sheet_name_to_unique_id_map(file_or_filename, lang):
+
+    def get_sheet_name():
+        return MODULES_AND_FORMS_SHEET_NAME if is_multisheet() else SINGLE_SHEET_NAME
+
+    def is_multisheet():
+        return lang is None
+
+    def is_modules_and_forms_row(row):
+        # Rows about the app's modules and forms, like their names and unique IDs
+        return not row['case_property'] and not row['list_or_detail'] and not row['label']
+
+    sheet_name_to_unique_id = {}
+
+    try:
+        worksheet = get_single_worksheet(file_or_filename, title=get_sheet_name())
+    except WorkbookJSONError:
+        return sheet_name_to_unique_id
+
+    if is_multisheet():
+        rows = worksheet
+    else:
+        rows = (row for row in worksheet if is_modules_and_forms_row(row))
+
+    for row in get_unicode_dicts(rows):
+        sheet_name = row.get('menu_or_form', '')
+        unique_id = row.get('unique_id')
+        if unique_id and sheet_name not in sheet_name_to_unique_id:
```

Comment:
if sheet_name is in sheet_name_to_unique_id, i think we should raise an exception instead since two modules/forms should not have the same id, instead of letting it pass.